refactor(ast_codegen): use doc comments instead of `insert!`

overlookmotel opened the pull request:
Avoid the `insert!` macro in AST codegen. Use doc comments starting with special symbol `@` instead.

* Before: `insert!("// plain comment");`
* After: `///@ plain comment`
* Or: `//!@ plain comment`

Either `///@` or `//!@` is converted to plain `//` in output.

`//!@` is legal in top-of-file position, which allows us to inline `#![allow(...)]` attributes, which in my opinion makes the generators a bit easier to read.

overlookmotel commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4777?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4777** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4777?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* **#4775** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4775?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#4774** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4774?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#4773** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4773?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @overlookmotel and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

overlookmotel commented:
@rzvxa The `@` symbol is a bit mysterious, so not ideal, but on balance I think it's preferable to the harder-to-read `insert!()`. What do you think?

rzvxa commented:
To be honest I don't like this change, It is too complicated for a simple thing.
I prefer the syntax noise over having too many symbols.

With that said, Feel free to merge as you wish.

overlookmotel commented:
> To be honest I don't like this change, It is too complicated for a simple thing. I prefer the syntax noise over having too many symbols.
> 
> With that said, Feel free to merge as you wish.

OK. My feeling is that it's worth it for making [generators/visit.rs](https://github.com/oxc-project/oxc/pull/4777/files#diff-de12458ea01221ddacb63a48b9635dcb20de889430eeb67952a0d8b489fdb813) clearer.

Since you've given me permission to merge, I'm going to do that.

overlookmotel committed:
refactor(ast_codegen): use doc comments instead of `insert!` (#4777)

Avoid the `insert!` macro in AST codegen. Use doc comments starting with special symbol `@` instead.

* Before: `insert!("// plain comment");`
* After: `///@ plain comment`
* Or: `//!@ plain comment`

Either `///@` or `//!@` is converted to plain `//` in output.

`//!@` is legal in top-of-file position, which allows us to inline `#![allow(...)]` attributes, which in my opinion makes the generators a bit easier to read.
